src/environmental/water_quality/surface_water_quality.py

```diff
@@ -0,0 +1,304 @@
+    in the existing database.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    catchment_area : gpd.GeoDataFrame
+        A GeoDataFrame representing the catchment area.
+
+    Returns
+    -------
+    gpd.GeoDataFrame
+        A GeoDataFrame containing surface water quality data from ECAN for the requested catchment area that is
+        not already present in the existing database.
+    """  # noqa: D400
+    # Retrieve existing surface water quality data from the database for the requested catchment area
+    water_quality_exist = get_surface_water_quality_from_db(engine, catchment_area)
+    # Fetch surface water quality data from ECAN for the requested catchment area
+    water_quality_new = get_surface_water_quality_with_geom(engine, catchment_area)
+    # Identify records in the new surface water quality data that are not present in the existing database
+    water_quality_not_in_db = pd.concat([water_quality_exist, water_quality_new]).drop_duplicates(keep=False)
+    return water_quality_not_in_db
+
+
+def store_surface_water_quality_to_db(engine: Engine, catchment_area: gpd.GeoDataFrame) -> None:
+    """
+    Fetch surface water quality data from ECAN for the specified catchment area and store it in the database.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    catchment_area : gpd.GeoDataFrame
+        A GeoDataFrame representing the catchment area.
+    """
+    # Define the table name for storing the surface water quality data
+    table_name = "surface_water_quality"
+    # Check if the table already exists in the database
+    if check_table_exists(engine, table_name):
+        # Retrieve surface water quality data from ECAN for the requested catchment area
+        # that is not already present in the existing database
+        water_quality_not_in_db = get_surface_water_quality_not_in_db(engine, catchment_area)
+        # Check if any new surface water quality data was retrieved
+        if water_quality_not_in_db.empty:
+            # If no new data is found, log a message indicating that no new data was found
+            log.info("No new surface water quality data found for the requested catchment area.")
+        else:
+            # If new data is found, add it to the relevant table in the database
+            log.info(f"Adding '{table_name}' for sites within the requested catchment area to the database.")
+            water_quality_not_in_db.to_postgis(table_name, engine, index=False, if_exists="append")
+            log.info(f"Successfully added '{table_name}' to the database.")
+    else:
+        # Fetch surface water quality data from ECAN for the requested catchment area
+        surface_water_quality = get_surface_water_quality_with_geom(engine, catchment_area)
+        # Check if any surface water quality data was fetched
+        if not surface_water_quality.empty:
```

Comment:
What if it is empty? If it's never empty, we don't need a check. If it is possible to be empty, should it log an error?

Reply:
I've made some fixes for this in case there are no sites or no water quality data for the specified catchment area.